### ISSUE TITLE
Checking only visible text/current page/current node/current section?

### ISSUE THREAD

**Issue by ghost** [PillFall/Emacs-LanguageTool.el#2]
Hello and a HUGE thank you for this great plug-in.

I have enjoyed it immensely, it is a giant help to people writing in Emacs.

I would like, though, to ask for a feature, if possible.

At the moment, the "automatic" checking mode, `languagetool-server-mode`, works on the whole buffer.
This is usually not required (although, I believe, there are checks that only work globally.), and is very resource-intensive.

Would it be possible to add an option to restrict server-mode checking to the visible text, perhaps launched automatically after a short period of inactivity, say, 2 seconds?

The authors of `spell-fu` https://gitlab.com/ideasman42/emacs-spell-fu implement this somehow, as an alternative to flyspell mode.

**Comment by PillFall** [comment]
Hello!

I kept that in mind when creating this plug-in, and it should only check the current accessible portion of the buffer, even when it's narrowed. Yet it have to wide it to clear all the previous suggestions in case of any.

I'll check that out, cause I know that the current system has a really poor performance. I'm trying to replicate ```flyspell``` behaviour (I didn't know about ```spell-fu```, I'll take a look).

And yes. I think that would be a great idea. I would implement that these days. (In the begining I have implemented that way, But was a mess. 😂)
